mandrel/test/bootstrap/bootstrap_file_test.py

```diff
@@ -43,3 +44,12 @@ def testDefaultLoggingConfig(self):
             utils.refresh_bootstrapper()
             self.assertEqual('logging.cfg', mandrel.bootstrap.LOGGING_CONFIG_BASENAME)
 
+    def testOSEnv(self):
+        with EnvironmentVarGuard() as env:
+            env.set('MANDREL_ROOT', '/blah')
+            env.set('MANDREL_BOOTSTRAP_NAME', 'bootstrapper.py')
+
+            utils.refresh_bootstrapper()
+            self.assertEqual(os.getenv('MANDREL_ROOT'), mandrel.bootstrap.ROOT_PATH)
+            expected = os.path.join(os.getenv('MANDREL_ROOT'), os.getenv('MANDREL_BOOTSTRAP_NAME'))
+            self.assertEqual(expected, mandrel.bootstrap.BOOTSTRAP_FILE)
```

Comment:
I'm happy to see tests, but this changes too much core stuff for a single test case to cover.

Given that you're introducing two changes (the mandrel root variable; the bootstrap basename variable), the cases to consider are considerably more complicated:
- Base case: neither variable is defined (the original behaviors)
- Root is specified, bootstrapper is not, `Mandrel.py` is present in specified root (should parse the file it finds)
- Root is specified, bootstrapper is not, `Mandrel.py` is _not_ present in specified root (should get basic defaults)
- Root is specified, bootstrapper is specified, the specified file exists in the specified root
- Root is specified, bootstrapper is specified, the specified file does not exist in the specified root
- Root is not specified, bootstrapper is specified, the specified file exists; it should be used and the root should be where that guy was found.  This would need testing at various file system levels.
- Root is not specified, bootstrapper is specified, the specified file does not exist within the file system hierarchy; this should blow up, just like it does now if `Mandrel.py` can't be found.

So there's quite a few test cases that would need to be covered.

It's considerably simpler if only one of the issues is done: either specifying a root, or specifying a basename, but not both.

But as it stands, the test only covers the case where both are set, it doesn't treat them separately.